### DISCUSSION TITLE
Pneunet Warning Determinant is null

### DISCUSSION THREAD

**Discussion by andersonnardin** [SofaDefrost/SoftRobots#143] [Help - Programming with SoftRobots] [answered]
Hello all,

I have a basic doubt. In the pneunet tutorial, there is a warning that appears constantly:
`[WARNING] [LocalMinDistance(Proximity)] Determinant is null`

The respective configuration is:
`rootNode.addObject('LocalMinDistance', name='Proximity', alarmDistance=5, contactDistance=1, angleCone=0.0)`

So, why is the warning appearing? How to avoid it?

Thank you in advance

**Comment by EulalieCoevoet** [comment, marked as answer]
When we look at the code it comes from `LocalMinDistance::computeIntersection()` between two lines:

```cpp
// several possibilities :
// -one point in common (auto-collision) => return false !
// -no point in common but line are // => we can continue to test
msg_warning() <<"Determinant is null";
```

I guess there might be cases of parallel lines as mentioned in the comments before the warning. 
We could argue that this warning sould be removed or at least give more information.

**Reply by andersonnardin** [reply]
Hi @EulalieCoevoet 

I agree that this warning is at least annoying because, as it is generated at each step, it disturbs the reading of other pieces of information.
So, I found a workaround to avoid this message by simply changing the rotation of the plane from 270 to 271 degrees:

`planeNode.addObject('MeshObjLoader', name='loader', filename='data/mesh/floorFlat.obj', triangulate=True,
                        rotation=[0, 0, 271], scale=10, translation=[-122, 0, 0])`

Anyway, thank you.

**Reply by EulalieCoevoet** [reply]
This warning has been improved (will be included in SOFA release v22.06).
